packages/flutter/lib/src/painting/inline_span.dart

```diff
@@ -0,0 +1,257 @@
+  /// ```dart
+  /// class BuzzingText extends StatefulWidget {
+  ///   @override
+  ///   _BuzzingTextState createState() => _BuzzingTextState();
+  /// }
+  ///
+  /// class _BuzzingTextState extends State<BuzzingText> {
+  ///   LongPressGestureRecognizer _longPressRecognizer;
+  ///
+  ///   @override
+  ///   void initState() {
+  ///     super.initState();
+  ///     _longPressRecognizer = LongPressGestureRecognizer()
+  ///       ..onLongPress = _handlePress;
+  ///   }
+  ///
+  ///   @override
+  ///   void dispose() {
+  ///     _longPressRecognizer.dispose();
+  ///     super.dispose();
+  ///   }
+  ///
+  ///   void _handlePress() {
+  ///     HapticFeedback.vibrate();
+  ///   }
+  ///
+  ///   @override
+  ///   Widget build(BuildContext context) {
+  ///     return RichText(
+  ///       text: TextSpan(
+  ///         text: 'Can you ',
+  ///         style: TextStyle(color: Colors.black),
+  ///         children: <InlineSpan>[
+  ///           TextSpan(
+  ///             text: 'find the',
+  ///             style: TextStyle(
+  ///               color: Colors.green,
+  ///               decoration: TextDecoration.underline,
+  ///               decorationStyle: TextDecorationStyle.wavy,
+  ///             ),
+  ///             recognizer: _longPressRecognizer,
+  ///           ),
+  ///           TextSpan(
+  ///             text: ' secret?',
+  ///           ),
+  ///         ],
+  ///       ),
+  ///     );
+  ///   }
+  /// }
+  /// ```
+  /// {@end-tool}
+  final GestureRecognizer recognizer;
+
+  /// An alternative semantics label for this text.
+  ///
+  /// If present, the semantics of this span will contain this value instead
+  /// of the actual text.
+  ///
+  /// This is useful for replacing abbreviations or shorthands with the full
+  /// text value:
+  ///
+  /// ```dart
+  /// TextSpan(text: r'$$', semanticsLabel: 'Double dollars')
+  /// ```
+  final String semanticsLabel;
+
+  /// Utility method to convert InlineSpan into [TextSpan] or [WidgetSpan].
+  ///
+  /// Will return null if span is not of type T or null. Otherwise, return
+  /// the span cast as T. Values of T that are not subclasses of [InlineSpan]
+  /// will result in null.
+  static T asType<T>(InlineSpan span) {
+    if (span is! T)
+      return null;
+    return span as T;
+  }
+
+  /// Apply the properties of this object to the given [ParagraphBuilder], from
+  /// which a [Paragraph] can be obtained. [Paragraph] objects can be drawn on
+  /// [Canvas] objects.
+  void build(ui.ParagraphBuilder builder, { double textScaleFactor = 1.0, List<PlaceholderDimensions> dimensions });
+
+  /// Walks this InlineSpan and any descendants in pre-order and calls [visitor]
+  /// for each span that has content.
+  ///
+  /// When [visitor] returns true, the walk will continue. When [visitor] returns
+  /// false, then the walk will end.
+  ///
+  /// The type of the span may be checked using [asType].
+  bool visitInlineSpan(bool visitor(InlineSpan span));
```

Comment:
we usually call this "visitChildren", and prefer to use a typedef for the argument. (this was bad in the original code too)

Reply:
Should I change this API to take a typedef-ed argument or just leave it as is for better backwards compatibility? I believe changing the type to InlineSpan already constitutes a minor breaking change.

Reply:
It won't break anything to use a typedef.

Reply:
Changing the name obviously would be a breaking change, but you've done that already, so everything else is academic.